### PR TITLE
Policy updates and Fix for bug #66

### PR DESCRIPTION
**Pull request by suchandanreddy**
Includes code for ability to change preferred color in application aware routing policy using the command `vmanage --host 10.10.10.10 --username admin --password pass modify-central-policy modify-approute-policy -n approute_policy_name -prefcolor public-internet`

contains fix for bug #66 to handle below usecases.
-  when we want to export attachments for all devices including devices which doesn't have host-name or site-id.
-  handle Variables which doesn't have user defined variable name in `()` and use <b>toolTip</b> instead of default `None` because that would override variables with `null` or `None` key when we have multiple variables which doesn't have user defined variable names within `()`

### CODE REVIEW

File: vmanage/cli/modify_central_policy/modify_approute_policy.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import click
+from vmanage.api.policy_updates import PolicyUpdates
+
+
+@click.command('modify-approute-policy')
+@click.option('--name', '-n', help="Approute policy name")
+@click.option('--pref_color', '-prefcolor', help="Preferred color")
```
Comment (jlothian) [inline vmanage/cli/modify_central_policy/modify_approute_policy.py:7]: Can you rename this to --pref-color to keep it in line with other long options?

**Comment by jlothian** [comment]
Another general comment is that going forward, it would be helpful to separate out bug-fixes and new features in separate PRs

File: vmanage/cli/modify_central_policy/modify_approute_policy.py (hunk 1 above)
Comment (suchandanreddy) [reply vmanage/cli/modify_central_policy/modify_approute_policy.py]: I have added the commands to file tests/commands.txt and required policy to the tests directory. Also changed the option to --pref-color. Please check and let me know if we need any further changes to merge this to master.

**Comment by jlothian** [comment]
@suchandanreddy those changes look fine to me.  I'd still like @jasonking3  to take a look since he's a lot more familiar with the actual code than I am.

**Comment by jasonking3** [comment]
@suchandanreddy - What do you think about using the "set" command and overloading it to do the policy updates?  Something like `vmanage set central-policy --type approute --name approute_policy_name --prefcolor public-internet`.  The reason I suggest this is that I get worried about a proliferation of base commands when really what you want to do here is set a value in the central policy.

**Comment by suchandanreddy** [comment]
@jasonking3 the idea I had was to use the option `modify-central-policy` to support options for updating various other parameters like updating sla class values , data policy actions etc.. I can definitely move these under set commands but only thing I am worried what we would be supporting with policies is more like an update to existing policy rather than setting a value.

**Comment by pm7625** [comment]
Hi,

is there option to attach device (with serial#) to specific template with vmanage command ? Import/export is to import/export templates with attachemnts(variables). But I couldnt find option how to attach device with variables to specific template. Such command I couldnt find.

**Comment by Azarack** [comment]
> Hi,
> 
> is there option to attach device (with serial#) to specific template with vmanage command ? Import/export is to import/export templates with attachemnts(variables). But I couldnt find option how to attach device with variables to specific template. Such command I couldnt find.

In the device templates class there is an attach to template call.  This takes in the variables needed to attach a specific device to a specified template.  Is that what you are looking for?

**Comment by suchandanreddy** [comment]
@jasonking3 I have moved the policy modify options to set command and now the new command examples are `vmanage set central-policy -t approute -n Outlook-AAR-Policy -prefcolor public-internet -seqname Outlook
` and `vmanage set central-policy -t approute -n Outlook-AAR-Policy -prefcolor biz-internet`. Could you please check and let me know if you need further changes to approve the PR.

**Comment by jasonking3** [comment]
@suchandanreddy I made some minor changes.

- I removed your pylint directive to ignore the name collision errors with built-ins and changed the `type` variable to `type_`.  This is the correct way to handle it according to PEP8.
- I removed the single-dash option for pref-color and seq-name since they didn't really save any typing over the double-dash options.  If we are going to have single-dash options, I'd like to see those be short (ideally one character).

Please let me know if you are OK with the changes and I'll merge this PR.

**Pull request merged**